src/vs/vscode.proposed.d.ts

```diff
@@ -1580,7 +1636,23 @@ declare module 'vscode' {
 			provider: NotebookProvider
 		): Disposable;
 
-		export function registerNotebookOutputRenderer(type: string, outputSelector: NotebookOutputSelector, renderer: NotebookOutputRenderer): Disposable;
+		export function registerNotebookExecution(
+			execution: NotebookExecution
```

Comment:
Looking at this API, who is responsible for executing cells?
VSC or extension author?
If its extension author then we need a way to provide results to VSC in response to an execute request or similar.

Reply:
VSC is responsible. THat's what the NotebookExecutionResult is for. It has events they have to listen to for output etc.